src/analysisd/decoders/decoder.h

```diff
@@ -79,18 +81,37 @@ typedef struct dbsync_context_t {
  * list and to get the first osdecoder
  */
 void OS_CreateOSDecoderList(void);
-int OS_AddOSDecoder(OSDecoderInfo *pi, OSDecoderNode **pn_osdecodernode, OSDecoderNode **npn_osdecodernode);
+int OS_AddOSDecoder(OSDecoderInfo *pi, OSDecoderNode **pn_osdecodernode, OSDecoderNode **npn_osdecodernode, char **msg);
 OSDecoderNode *OS_GetFirstOSDecoder(const char *pname);
 int getDecoderfromlist(const char *name);
 char *GetGeoInfobyIP(char *ip_addr);
-int SetDecodeXML(void);
+/**
+ * @brief Add decoders of base modules to main list.
+ * @param[out] msg If any error or warning occurs, your description are stored in *msg.
```

Comment:
Can we do it briefer and remove the tabs?

Reply:
Maybe something like this?
```
/**
 * @brief Read decoder files and save them in the decoder list.
 * @param[out] msg Store warnings and error as result of call
 * @return int 0 in case of error, 1 otherwise.
 * @note If *msg==null, memory are allocate, otherwise memory are reallocate.
 */
```

Reply:
yes